File: Resources/Archive-2015/6. AppLifecycleDemos/AppLifecycleDemos/AppLifecycleDemos/ViewController.m
//
//  ViewController.m
//  AppLifecycleDemos
//
//  Created by Doncho Minkov on 10/30/14.
//  Copyright (c) 2014 Telerik Academy. All rights reserved.
//

#import "ViewController.h"
#import "AppDelegate.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
//    UIButton *button = [UIButton buttonWithType:UIButtonTypeRoundedRect];
//    
//    [button setTitle:@"Click" forState:UIControlStateNormal];
//    [button setTitle:@"Clicked" forState:UIControlStateHighlighted];
//    button.backgroundColor = [UIColor yellowColor];
//    
//    
//    button.frame = CGRectMake(50, 50, 150, 170);
//    
//
//    UIView* subview = (UIView*) button;
//    [self.view addSubview:subview];
    
    
    
    
    
//    AppDelegate *app = (AppDelegate*)[[UIApplication sharedApplication] delegate];
//    NSString* namesList = [app.names componentsJoinedByString:@", "];
//    NSLog(@"%@", namesList);
    
    
    
//    [app.everlive data:@"Persons"]
}

-(void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
    
    NSLog(@"View will appear");
}

-(void)viewDidAppear:(BOOL)animated{
    [super viewDidAppear:animated];
    
    NSLog(@"View did appear");
    
}

-(void)viewWillDisappear:(BOOL)animated{
    [super viewWillDisappear:animated];
    
    NSLog(@"View will disappear");
    
}

-(void)viewDidDisappear:(BOOL)animated{
    [super viewDidDisappear:animated];
    
    NSLog(@"View did disappear");
    
}

-(void)viewWillLayoutSubviews{
    [super viewWillLayoutSubviews];
    
    NSLog(@"View will layout subviews");
    
}

-(void)viewDidLayoutSubviews{
    [super viewDidLayoutSubviews];
    
    NSLog(@"View did layout subviews");
}

-viewDId

-(IBAction) returnToMainView: (UIStoryboardSegue*)segue{
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: demos/MvcDemos/MvcDemos/ViewController.swift
//
//  ViewController.swift
//  MvcDemos
//
//  Created by Doncho Minkov on 3/24/17.
//  Copyright © 2017 Doncho Minkov. All rights reserved.

import UIKit

class MvcView:UIView{
    var delegate: BaseMvcController?
    
    func click() {
//        self.delegate.clickFired()
    }
}

class MvcController {
    var view: MvcController?
}

class ViewController: UIViewController {
    var mvcController: MvcController?
    
    override func viewDidLoad() {
        super.viewDidLoad()
//        self.mvcView = MvcView(frame: self.view.frame)
        self.mvcController = MvcController(withView: self.view)
        self.view.addSubview(self.view!)
        //if on wide screen
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewWillAppear(_ animated: Bool) {
    }
}

